### ISSUE TITLE
Real-time scanning doesn't refresh for whitespace characters

### ISSUE THREAD

**Issue by marcinkunert** [jshiell/checkstyle-idea#513]
IntelliJ IDEA Ultimate 2020.3
CheckStyle-IDEA 5.46.0
Checkstyle version: 8.36.2

I've chosen Google Checks, but the same happens to custom checkstyle config file. 

Simple file as example:

```
public class Main {

  void method1() {

  }

  void method2() {
    
  }

}
```

Steps to reproduce: 
1. Move cursor before the second method declaration and press backspace. No tooltip is visible. 
2. Remove or add any letter to the code and add it back. The tooltip will be shown.

Looks like the check is not triggered by 'empty' characters (enter, space, tab).

I've recorded a git to show it a bit better.
![checkstyle-issue](https://user-images.githubusercontent.com/2963232/104461331-8392b600-55af-11eb-9306-aed08c710518.gif)

I've enabled the logging and surprisingly it seems that the checks are being run after adding whitespace characters, but they are not detecting any issues.

```
2021-01-13 15:08:09,718 [2147377]  DEBUG - heckstyle.CheckStyleInspection - Inspection cancelled when scanning: Main.java 
2021-01-13 15:08:10,037 [2147696]  DEBUG - heckstyle.CheckStyleInspection - Inspection has been invoked for Main.java 
2021-01-13 15:08:10,040 [2147699]  DEBUG - ckstyle.checker.CheckerFactory - Getting CheckStyle checker with location Google Checks 
2021-01-13 15:08:10,040 [2147699]  DEBUG - le.checker.CheckerFactoryCache - Valid cached checker found; returning for Google Checks 
```

**Comment by MariuszSzmagara** [comment]
👍

**Comment by jshiell** [comment]
Thanks for the great report.

Bad news - we rely on IDEA for the trigger - I'll double check the integration when I get a moment, but I don't believe there's a lot of option here for fine-tuning.

https://github.com/jshiell/checkstyle-idea/blob/09353674adffb5f73630eeafca5d6f359da53599/src/main/java/org/infernus/idea/checkstyle/CheckStyleInspection.java#L55

https://github.com/jshiell/checkstyle-idea/blob/09353674adffb5f73630eeafca5d6f359da53599/src/main/resources/META-INF/plugin.xml#L96

**Comment by marcinkunert** [comment]
@jshiell I've edited the issue and added debug log output: the trigger seems to be firing, but the check is not reporting any issues ultil a non-whitespace character is added or removed

**Comment by jshiell** [comment]
Ah, that's more interesting. My guess at first glance is that it could be something to do with the file read - we need to make a copy of the file contents to scan, as Checkstyle needs a static file on disk (and IDEA may not have written changes). If  there's something awry with the syncing there on whitespace (e.g. IDEA doesn't flag the change) then we could well end up scanning something unexpected.

**Comment by jshiell** [comment]
Good news - I _finally_ had a look at it.

Bad news - it appears to be an IDEA issue. We return exactly the same `ProblemDescriptor`s from the inspection in both cases, but IDEA seems to ignore them in the whitespace case.

I've created an issue with Youtrack - https://youtrack.jetbrains.com/issue/IDEA-265941

**Comment by jshiell** [comment]
It turns out there's a scary attribute to turn on whitespace scanning, with the appropriate perf impact. That's now active as of 5.52.0, which should resolve this problem.
